Feature/kan 11/구인정보 api 개발

Hyeonjiwon opened the pull request:
- 채용정보 검색 API 구현 (kan-24)
- API 비동기 처리 및 페이지네이션 추가 (kan-36)

Hyeonjiwon committed:
[feat] 채용정보 검색 API 구현 (kan-24)

Motivation:
- 사용자가 조건으로 채용 정보를 검색할 수 있도록 API 엔드포인트 추가
- FastAPI에서 검색 조건에 따라 MongoDB에서 채용 정보를 필터링하는 API 구현
- '/job_posts/search' 엔드포인트 추가 및 검색 로직 구현.

Modification:
- 'job_post.py' 수정

Hyeonjiwon committed:
Merge branch 'feature/kan-11/구인정보-API-개발' of https://github.com/G-E-ANADA/inabe-be into feature/kan-11/구인정보-API-개발

Hyeonjiwon committed:
[feat] API 비동기 처리 및 페이지네이션 추가 (kan-36)

Motivation:
- MongoDB 쿼리 및 페이지네이션을 비동기로 처리하여 성능 개선 및 응답 속도 향상

Modification:
- `search_job_posts` 및 `list_job_posts` 엔드포인트를 비동기 방식으로 수정
- `total_count` 계산 및 페이지네이션 로직 추가
- API 응답 및 에러 처리 개선
- API 설명 업데이트

cogito1016 commented on api/job_post.py:
```diff
@@ -26,27 +21,30 @@
 if not origins:
     origins = ["http://localhost:4000",
-               "http://localhost:1234"]  # Default origins
-
+               "http://localhost:1234",
+               "http://localhost:53673"]  # Default origins
```
추후에는 환경변수처리가 필요해보입니다 :)

Hyeonjiwon replied:
생각하지 못하고 있던 부분이었는데  동의합니다 :)

cogito1016 commented on api/job_post.py:
```diff
@@ -88,25 +86,84 @@ class JobPostModel(BaseModel):
+    await check_mongo_connection()
+
+
 @app.get(
     "/job_posts/",
     response_description="List all job posts",
     response_model=JobPostCollection,
     response_model_by_alias=False,
 )
```
페이지네이션없이 모든 채용정보를 갖고오는건가요?
만약그렇다면 어떤용도로 해당 API가 쓰일 수 있을까요?

Hyeonjiwon replied:
"/job_posts/", API는 검색 API 개발 전 테스트 용도로 만든 전체 데이터가 호출되는 API 입니다. 현재 화면에서는 사용하지 않고있습니다. 혹시 몰라 남겨뒀는데 추후에 불필요하다고 판단되면 삭제 하겠습니다!

cogito1016 commented on api/job_post.py:
```diff
@@ -88,25 +86,84 @@ class JobPostModel(BaseModel):
+    total_count = await job_post_collection.count_documents(query)
+
+    # 페이지네이션 적용
+    skip = (page - 1) * limit
+    job_posts_cursor = job_post_collection.find(query).skip(skip).limit(limit)
+    job_posts = await job_posts_cursor.to_list(length=limit)
```
페이지네이션을 해당 API뿐만아니라, 서버 내 다른 API에서도 쓰일 가능성이있다면
공통 라이브러리로 관리해두는것도 고려해볼만할것같습니다.

Hyeonjiwon replied:
네 나중에 공통 라이브러리로 관리하는 법에 대해 상의해보시죠.

Hyeonjiwon committed:
[fix] 검색 API 페이징 처리 수정

Motivation:
- 페이지 이동 시 잘못된 데이터가 반환되는 문제를 해결

Modification:
- 'search_job_posts' 함수에서 'start' 파라미터를 'skip'으로 활용하여 페이징 처리 로직 수정
- 쿼리 파라미터와 페이지네이션 관련 로직 수정.
